__init__.py

```diff
@@ -70,183 +71,99 @@ def parse(self, utterance):
         return None
 
 
-class WAApi(Api):
-    def __init__(self):
-        super(WAApi, self).__init__("wa")
-
-    def get_data(self, response):
-        return response
-
-    def query(self, input):
-        data = self.request({"query": {"input": input}})
-        return wolframalpha.Result(StringIO(data.content))
-
-
 class WolframAlphaSkill(FallbackSkill):
-    PIDS = ['Value', 'NotableFacts:PeopleData', 'BasicInformation:PeopleData',
-            'Definition', 'DecimalApproximation']
 
     def __init__(self):
         FallbackSkill.__init__(self, name="WolframAlphaSkill")
-        self.__init_client()
-        self.question_parser = EnglishQuestionParser()
 
-    def __init_client(self):
-        key = self.config.get('api_key')
-        if key and not self.config.get('proxy'):
-            self.client = wolframalpha.Client(key)
-        else:
-            self.client = WAApi()
+        # Support 'api_key' in the old mycroft.conf location
+        self.api_key = self.config.get('api_key')
+
+        # TODO: Localization support for questions in other languages
+        self.question_parser = EnglishQuestionParser()
 
     def initialize(self):
-        self.init_dialog(dirname(__file__))
         self.register_fallback(self.handle_fallback, 8)
 
-    def get_result(self, res):
-        try:
-            return next(res.results).text
-        except:
-            result = None
-            try:
-                for pid in self.PIDS:
-                    result = self.__find_pod_id(res.pods, pid)
-                    if result:
-                        result = result[:5]
-                        break
-                if not result:
-                    result = self.__find_num(res.pods, '200')
-                return result
-            except:
-                return result
-
-    # TODO: Localization
     def handle_fallback(self, message):
+        if not self.api_key:
+            # attempt to get from webUI
+            self.api_key = self.settings['api_key']
```

Comment:
Should be `.get('api_key')` or
```Python
if 'api_key' not in self.settings:
    self.speak_dialog()#...
self.api_key = self.settings['api_key']
```
since otherwise it excepts rather than speaking the dialog.

Reply:
Hmm, I thought SkillSettings showed this behavior.  Tweaked to use .get()